datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -286,6 +287,39 @@ pub fn extract_projection(
     }
 }
 
+fn ensure_schema_match(
+    table: DataFrame,
+    expected: &NamedStruct,
+    extensions: &Extensions,
+) -> Result<DataFrame> {
+    // Substrait schema contains no qualifiers, so we need to strip them from the DF schema as well
+    // for the comparison to work
+    let df_schema = table.schema().to_owned().strip_qualifiers();
```

Comment:
Substrait does include qualifiers, I think. A NamedTable includes the table name which I believe should be used as the qualifier for all of the fields in that table. This is what I was attempting to handle in [`a44a689` (#12245)](https://github.com/apache/datafusion/pull/12245/commits/a44a689c955ee295558752c82d4ddbcc597336ef)